### PR TITLE
Store hmaced partner api tokens

### PR DESCRIPTION
**Pull request by cquintana92**
This PR performs the following changes:

1. Add a migration for increasing the size of partner api tokens
2. Add a `PartnerApiToken.generate` method that takes care of generating api tokens, processing them and storing them into the database
3. Add a `Partner.find_by_token` method that allows retrieving a `Partner` instance given a raw `PartnerApiToken`.

### CODE REVIEW

File: app/models.py [hunk 1]
```diff
@@ -3075,16 +3082,55 @@ class Partner(Base, ModelMixin):
...
+            Session.query(Partner, PartnerApiToken)
+            .filter(
+                and_(
+                    PartnerApiToken.token == hmaced,
+                    Partner.id == PartnerApiToken.partner_id,
+                )
+            )
+            .first()
+        )
+        if res is None or res.Partner is None:
```
Comment (nguyenkims) [inline app/models.py:3098]: Oh I didn't know we can use `res.Partner` here. Usually the result is unpacked with `partner, api_token=res`

**Review by nguyenkims — APPROVED** [review]
LGTM!

File: tests/models/test_partner_api_token.py [hunk 2]
```diff
@@ -0,0 +1,25 @@
...
+
+
+def test_generate_partner_api_token(flask_client):
+    partner = Partner.create(
+        name=random_string(10),
+        contact_email="{s}@{s}.com".format(s=random_string(10)),
+        commit=True,
+    )
+
+    (partner_api_token, token) = PartnerApiToken.generate(partner.id, None)
```
Comment (nguyenkims) [inline tests/models/test_partner_api_token.py:12]: Nits: we can remove parenthesis `(` here :)

Comment (cquintana92) [reply tests/models/test_partner_api_token.py]: Removed

File: app/models.py (hunk 1 above)
Comment (cquintana92) [reply app/models.py]: Changed it

Comment (cquintana92) [reply app/models.py]: Sorry, re-reading the comment I have noticed I misunderstood. Fixed the `None` crash and replaced it with the destructuring syntax

Comment (acasajus) [reply app/models.py]: Either keep it as it is or check for res not to be None before unpack. Otherwise it will fail :)

**Pull request merged**